spec/controllers/admin/become_controller_spec.rb

```diff
@@ -1,41 +1,41 @@
 require 'spec_helper'
 
-describe Admin::BecomeController do
+describe Admin::BecomeController, :type => :controller do
   let(:admin) { FactoryGirl.create(:admin) }
-  let(:user) { FactoryGirl.create(:user) }
-  let(:other_user) { FactoryGirl.create(:user) }
-  
+  let(:user) { FactoryGirl.create(:jill) }
+  let(:other_user) { FactoryGirl.create(:alice) }
+
   context 'admin' do
     before do
       sign_in admin
     end
 
     it 'should become user' do
-      get :index, id: user.email, use_route: :admin
```

Comment:
Controller specs are unit test. They should have zero dependency on how the routes are working. I removed everywhere we are using `use_route`